spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/ftp/FtpSessionFactoryConfigurationProperties.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.ftp;
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.constraints.Range;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author David Turanski
+ */
+@ConfigurationProperties
+public class FtpSessionFactoryConfigurationProperties {
+
+	/**
+	 * The host name of the FTP server. Default is 'localhost'. 
+	 */
+	private String host = "localhost";
+
+	/**
+	 * The port of the FTP server. Default is 21.
+	 */
+	private int port = 21;
+
+	/**
+	 * The username to use to connect to the FTP server. 
+	 */
+	private String username;
+
+	/**
+	 * The password to use to connect to the FTP server. 
+	 */
+	private String password;
+
+	/**
+	 * The client mode to use for the FTP session. Default is 0.
+	 */
+	private int clientMode = 0;
+
+	@NotBlank
+	public String getHost() {
+		return host;
+	}
+
+	public void setHost(String host) {
+		this.host = host;
+	}
+
+	@Range(min = 0, max = 65535)
+	public int getPort() {
+		return port;
+	}
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	public String getUsername() {
```

Comment:
Even when adding  `@NotBlank` the validation doesn't seem to kick in 

```
java.lang.IllegalArgumentException: 'user' should be a nonempty string
	at org.springframework.util.Assert.hasText(Assert.java:168) ~[spring-core-4.2.1.BUILD-SNAPSHOT.jar!/:4.2.1.BUILD-SNAPSHOT]
	at org.springframework.integration.ftp.session.AbstractFtpSessionFactory.setUsername(AbstractFtpSessionFactory.java:111) ~[spring-integration-ftp-4.2.0.BUILD-SNAPSHOT.jar!/:na]
	at org.springframework.cloud.stream.module.ftp.FtpSessionFactoryConfiguration.ftpSessionFactory(FtpSessionFactoryConfiguration.java:40) ~[spring-cloud-stream-modules-common-configuration-1.0.0.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.stream.module.ftp.FtpSessionFactoryConfiguration$$EnhancerBySpringCGLIB$$2b07175a.CGLIB$ftpSessionFactory$0(<generated>) ~[spring-core-4.2.1.BUILD-SNAPSHOT.jar!/:1.0.0.BUILD-SNAPSHOT]
```
Not sure why.  Also, should we put the JSR303 annotation on the field declaration?